Automattic/wp-calypso
# Build: Use file-loader to load flag SVG

Alternative to #26413.

Use webpack's `file-loader` to copy flag SVG assets to `public/images` when it encounters code that `require()` them. This allows us to drop the unconditional copying we previously did through `CopyWebpackPlugin`, and the code that disabled it for the SDK. Henceforth, webpack will be able to determine on its own whether to include flags SVGs when bundling, depending on whether there's a `require()` that calls for them.

Note we're now copying to `public/images/` rather than `public/images/flags/`, since the `file-loader` is generic enough and agnostic of flags to be able to copy any kind of asset whose extension we whitelist -- currently that's only `.svg`, but doesn't mean it's always going to be just flags. 

In order not to break tests (or have them unduly depend on webpack loaders), we're also adding a Jest helper that essentially translates `require( 'path/to/some.svg' )` to `'some.svg'` (h/t @simison for pointing me to https://jestjs.io/docs/en/webpack#handling-static-assets).


Resolves #26177 #26296

To test:
* `npm run distclean && npm i && npm start`
* Check phone input dropdown ('Form Media Phone Input') at http://calypso.localhost:3000/devdocs/design/form-fields
* Check the per-country stats view at `http://calypso.localhost:3000/stats/day/countryviews/:site:`. Verify that there are flag icons next to each country's name ([see](https://github.com/Automattic/wp-calypso/pull/26555#pullrequestreview-144707436)).
* Verify that after ending Calypso, no flag assets are in `flags/images/*`. Try a wpcalypso build in Docker to be sure. (#26141 thus being resolved)
  * You can try more explicitly by running `npm run build-client` and verifying that flag SVGs are created in `public/images/`, followed by `npm run clean` and verifying that they're removed.
